jujugui/static/gui/src/app/components/profile/bundle-list/test-bundle-list.js

```diff
@@ -0,0 +1,122 @@
+/* Copyright (C) 2017 Canonical Ltd. */
+
+'use strict';
+
+var juju = {components: {}}; // eslint-disable-line no-unused-vars
+
+describe('Profile Bundle List', function() {
+
+  const rawBundleData = `[{
+    "id": "cs:~lazypower/bundle/logstash-core-1",
+    "perm": {
+      "read": ["everyone"],
+      "write": ["lazypower"]
+    },
+    "applications": {
+      "elasticsearch": {
+        "charm": "cs:~lazypower/trusty/elasticsearch"
+      },
+      "kibana": {
+        "charm": "cs:trusty/kibana-10"
+      },
+      "logstash": {
+        "charm": "cs:~lazypower/trusty/logstash-20"
+      },
+      "openjdk": {
+        "charm": "cs:~kwmonroe/trusty/openjdk"
+      }
+    },
+    "name": "logstash-core",
+    "unitCount": 3
+  }, {
+    "id": "cs:~lazypower/bundle/swarm-core-1",
+    "perm": {
+      "read": ["lazypower", "everyone"],
+      "write": ["lazypower"]
+    },
+    "applications": {
+      "consul": {
+        "charm": "cs:~containers/trusty/consul"
+      },
+      "swarm": {
+        "charm": "cs:~lazypower/swarm"
+      }
+    },
+    "name": "swarm-core",
+    "unitCount": 5
+  }]`;
+
+  beforeAll(function(done) {
+    // By loading this file it adds the component to the juju components.
+    YUI().use('profile-bundle-list', function() {
+      done();
+    });
+  });
+
+  function renderComponent(options={}) {
+    const charmstoreList = (user, cb) => {
+      assert.equal(user, 'lazypower@external');
+      cb(null, JSON.parse(rawBundleData));
+    };
+    return jsTestUtils.shallowRender(
+      <juju.components.ProfileBundleList
+        addNotification={sinon.stub()}
+        baseURL="/gui/"
+        changeState={options.changeState || sinon.stub()}
+        charmstore={{
+          list: charmstoreList,
+          url: '/charmstore'
+        }}
+        user="lazypower@external" />, true);
+  }
+
+  it('can render', () => {
```

Comment:
Definitely, once there is a design, not much point in implementing tests that'll be deleted shortly.